src/runner/test-run/index.js

```diff
@@ -171,6 +197,33 @@ ServiceMessages[COMMAND.nativeDialogsInfoSet] = function (msg) {
     }
 };
 
-ServiceMessages[COMMAND.takeScreenshot] = function () {
-    //TODO:
+ServiceMessages[COMMAND.takeScreenshot] = async function (msg) {
+    await ensureDir(this.screenshotPath);
+
+    var pageUrl      = msg.pageUrl;
+    var stepName     = msg.stepName;
+    var isFailedStep = msg.isFailedStep;
+
+    var filePath = msg.filePath ?
+                   path.join(path.dirname(this.test.fixture.path), msg.filePath) :
+                   `${this.screenshotPath}\\${isFailedStep ? 'errors\\' : ''}${stepName.replace(/\s/g, '_') ||
+                                                                               'Page_Load'}.png`;
+
+    //NOTE: the test should not fail if we can't make a screenshot for some reason
+    try {
+        await takeScreenshot(pageUrl, filePath);
+
+        if (isFailedStep) {
```

Comment:
Let's make screenshot in `fatalError` and `assertionFailed` service messages if this functionality is enabled, instead of making everything in this service message.